includes/woopay/class-woopay-utilities.php

```diff
@@ -125,7 +125,7 @@ public function should_save_platform_customer() {
 			$session_data = WC()->session->get( WooPay_Session::WOOPAY_SESSION_KEY );
 		}
 
-		return isset( $session_data['save_user_in_woopay'] ) && filter_var( $session_data['save_user_in_woopay'], FILTER_VALIDATE_BOOLEAN );
+		return ( isset( $_POST['save_user_in_woopay'] ) && filter_var( wp_unslash( $_POST['save_user_in_woopay'] ), FILTER_VALIDATE_BOOLEAN ) ) || ( isset( $session_data['save_user_in_woopay'] ) && filter_var( $session_data['save_user_in_woopay'], FILTER_VALIDATE_BOOLEAN ) ); // phpcs:ignore WordPress.Security.NonceVerification
```

Comment:
🤏 How about splitting this into variables so it's easier to read? It took me a bit to understand how the conditionals were grouped. Thinking that it'd be simpler if we could do `return $is_save_user_in_post || $is_save_user_in_session`. Disregard the variable names, just referring to readability.

Not a blocker, though!